app/src/features/apiClient/helpers/APIClientManager/index.ts

```diff
@@ -7,26 +7,29 @@ import { renderTemplate } from "backend/environment/utils";
 import { DEMO_API_URL } from "features/apiClient/constants";
 import { trackAPIRequestSent } from "modules/analytics/events/features/apiClient";
 import { isEmpty } from "lodash";
-import { processAuthOptions, updateRequestWithAuthOptions } from "../auth";
+import { processAuthForEntry, updateRequestWithAuthOptions } from "../auth";
 
 export const executeAPIRequest = async (
   appMode: string,
+  apiRecords: RQAPI.Record[],
   entry: RQAPI.Entry,
+  entryDetails: {
+    id: RQAPI.Record["id"];
+    collectionId: RQAPI.Record["collectionId"];
+  },
```

Comment:
can this be prevented from passing? we can create this from entry.

Reply:
this isn't part of `Entry`. It is part of the `Record`. Here's a reference to the [type defination for this](https://github.com/requestly/requestly/blob/master/app/src/features/apiClient/types.ts#L80-L131). 

It is clear that these are required to know what is the id of the record as well as the parent record